lib/main.js

```diff
@@ -55,14 +67,56 @@ export default {
           });
         }
       }),
-    );
-
-    this.subscriptions.add(
       atom.config.observe('linter-tslint.useLocalTslint', (use) => {
         tslintCache.clear();
         this.useLocalTslint = use;
       }),
     );
+
+    atom.workspace.observeActivePaneItem((activeItem) => {
+      if (activeItem && activeItem.getGrammar) {
+        if (isTypescriptGrammar(activeItem.getGrammar())) {
```

Comment:
Went with a [solution](https://atom.io/docs/api/v1.15.0/TextEditor#instance-getRootScopeDescriptor) i found in the Atom api that seemed simpler.